core/src/com/cloud/agent/api/CancelMigrationCommand.java

```diff
@@ -0,0 +1,35 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+package com.cloud.agent.api;
+
+public class CancelMigrationCommand extends Command {
```

Comment:
What you're describing is something slightly different than the case I want to address with my command. Your concern is valid, and the hook should be added too. I've sent once an email to the mailing list to ask why an agent that disconnects cannot reconnect to a management server until all its tasks are done, because the command will always be marked as failed on the management server and might have been executed successfully on the agent, causing incoherent states.

For the migration, without the cancellation command, if the migration takes a long time, the management server will cancel the job after a while and mark it as failed, thus invalidating the new host information. But the agent will keep on migrating the VM and when it has moved to the new host it becomes unreachable/unavailable since the network information hasn't been updated/committed.